ui/src/lib/codeiumClient.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  Code,
+  ConnectError,
+  PromiseClient,
+  createPromiseClient,
+} from "@bufbuild/connect";
+import { createConnectTransport } from "@bufbuild/connect-web";
+import { PartialMessage } from "@bufbuild/protobuf";
+import { v4 as uuidv4 } from "uuid";
+import { Metadata } from "../proto/exa/codeium_common_pb/codeium_common_pb";
+import { LanguageServerService } from "../proto/exa/language_server_pb/language_server_connect";
+import {
+  AcceptCompletionRequest,
+  GetCompletionsRequest,
+  GetCompletionsResponse,
+} from "../proto/exa/language_server_pb/language_server_pb";
+
+const EXTENSION_NAME = "chrome";
+const EXTENSION_VERSION = "1.2.18";
+const BASE_URL = "https://server.codeium.com";
+
+function languageServerClient(): PromiseClient<typeof LanguageServerService> {
+  const transport = createConnectTransport({
+    baseUrl: BASE_URL,
+    useBinaryFormat: true,
+  });
+  return createPromiseClient(LanguageServerService, transport);
+}
+
+export interface IdeInfo {
+  ideName: string;
+  ideVersion: string;
+}
+
+export class LanguageServerClient {
+  private requestId = 0;
+  private sessionId = uuidv4();
+
+  client: PromiseClient<typeof LanguageServerService> = languageServerClient();
+  apiKey: string;
+
+  private abortController?: AbortController;
+
+  constructor(apiKey: string) {
+    this.apiKey = apiKey;
+  }
+
+  getHeaders(apiKey: string | undefined): Record<string, string> {
+    if (apiKey === undefined) {
+      return {};
+    }
+    const Authorization = `Basic ${apiKey}-${this.sessionId}`;
+    return { Authorization };
+  }
+
+  async getCompletions(
+    request: GetCompletionsRequest
+  ): Promise<GetCompletionsResponse | undefined> {
+    this.abortController?.abort();
+    this.abortController = new AbortController();
+    const signal = this.abortController.signal;
+    try {
+      const getCompletionsPromise = this.client.getCompletions(request, {
+        signal,
+        headers: this.getHeaders(request.metadata?.apiKey),
+      });
+      console.log("codeium request", request);
```

Comment:
This is printed very often. Change it to `console.debug` so that user's console stays clean.

Reply:
Or we could show an indicator in a "status bar" (or the toolbar near the "auto-complete" switch) to show that we are doing a Codeium request.

Reply:
I like the idea of indicator. Do we have to implement it in this PR? I can add it later.

Reply:
Sg, let’s add it later.